src/blog/2024/04/understanding-node-flow-global-environment-variables-in-node-red.md

```diff
@@ -0,0 +1,233 @@
+
+!["Screenshot showing how to retrieve environment variable in the change node"](./images/variables-in-node-red-retrieving-environment-variable-using-change-node.png "Screenshot showing how to retrieve environment variable in the change node"){data-zoomable}
+
+To access environment variables in a function node, use:
+
+```javascript
+env.get('variableName');
+```
+
+In the template node, you can access it like below:
+
+```javascript
+This is my username : {% raw %}{{env.USERNAME}}.
+{% endraw %}
+```
+
+And if you need to access environment variables in third-party configuration nodes where they haven't provided an option to use environment variables, you can access them with `${variableName}` in the node's input field.
+
+*Note: When you have variables with the same name, accessing them will prioritize the flow-level variable over the global-level one. To access a global-level environment variable in this scenario, you need to add a prefix to the variable name with '$parent'*
+
+#### Deleting Environment variables 
+
+To delete added environment variables, you can use the same interface where added them. In the right corner of your added environment variable, you'll see a delete or cross icon. Simply click on it to delete the variable and restart your Node-RED instance.
+
+### Exploring Context Data tab
+
+!["Screenshot showing Node-RED Context data tab"](./images/variables-in-node-red-context-data-tab.png "Screenshot showing Node-RED Context data tab"){data-zoomable}
+
+Node-RED provides a dedicated interface for viewing and managing all Node-RED variables. Navigate to the sidebar's "Context Data" tab, where you'll find sections for Node, Flow, and Global variables. Each section has a refresh icon at the top right corner; click on it to see the latest or newly added variables.
+
+!["Screenshot showing Node-RED Context data tab options for managing variables"](./images/variables-in-node-red-change-node-persistent-store-option.gif "Screenshot showing Node-RED Context data tab options for managing variables"){data-zoomable}
+
+In this tab, you'll also find information about when each variable was set or updated, along with additional options on the right side of each variable. The first option allows you to copy the variable's name, the second option lets you copy the variable's value, the third option refreshes the variable to show the most recent value, and the fourth option allows you to delete the variable.
+
+### FlowFuse persistent storage
+
+So far, we have seen three types of context variables and all are stored in memory. This means that when we stop the Node-RED instance, the variables will get wiped out.
+
+FlowFuse provides a way to persist these variable values between restarts and FlowFuse stack updates. For more details, refer to [FlowFuse Persistent Context](https://flowfuse.com/docs/user/persistent-context/).
```

Comment:
Internal links should not have the domain name first, now the link works when developing locally too.
```suggestion
FlowFuse provides a way to persist these variable values between restarts and FlowFuse stack updates. For more details, refer to [FlowFuse Persistent Context](/docs/user/persistent-context/).
```